### ISSUE TITLE
WCAG accessibility support

### ISSUE THREAD

**Issue by renniebeach** [RobinHerbots/Inputmask#981]
Would it be possible to have some aria support built into the plugin?

e.g.
one of the main issues I have come across using this plugin when trying to make a site WCAG compliant, it that when setting focus to an input field with a numeric mask applied, when using a screen reader the mask is announced "underline underline underline underline underline underline underline underline underline underline"

**Comment by RobinHerbots** [comment]
@renniebeach , 

Some extra info: 
http://www.w3.org/TR/wai-aria/states_and_properties#state_prop_def

I can add:

aria-invalid
aria-readonly
aria-valuemax
aria-valuemin
aria-valuetext

Any suggestions in particular?

**Comment by gunnx** [comment]
@renniebeach Use another solution like https://github.com/filamentgroup/politespace ?

**Comment by RobinHerbots** [comment]
@gunnx , 

That is a not so polite suggestion ;-))

**Comment by gunnx** [comment]
@RobinHerbots Apologies not meant to be negative, I've been trying to find something that meets this requirement too and so far I can't see a way around except looking at an alternative solution. Screen reader testing reveals that they read out the mask which can be quite confusing.

**Comment by RobinHerbots** [comment]
@gunnx , 

I was just joking ;-)

Anyway, I want to support WCAG, but I have no experience with it.  Any feedback is more then welcome or any suggestion to software to test it.  
Will the area attributes give the desired result?

**Comment by RobinHerbots** [comment]
@renniebeach , @gunnx , 

I did some tests with NVDA, and this reader does not  announce the placeholders (underline)
Is this also currently the case with other screen readers?  I could add the aria-label attribute with a description of the applied mask if that is any good?

**Comment by recrit** [comment]
@RobinHerbots I was able to do a temporary work around by setting `showMaskOnFocus : false`. With that set, the Apple VoiceOver no longer says "underscore, underscore' when you are navigating through the form.

**Comment by RobinHerbots** [comment]
@recrit , 

Thx for the feedback.  
Can you do some little test for me?
If you add aria-valuetext="Hello" to the input element does the Apple VoiceOver says the mask or does it say hello?

**Comment by stuartclan** [comment]
@RobinHerbots, take this testing for what its worth...

I played with VoiceOver on Mavericks & Sierra (don't have High Sierra near me right now) and used Chrome's inspector to add the aria-valuetext parameter you suggested.

That test didn't do anything - VoiceOver continued to read the input's contents with repeated underlines (using the Phone field as a test). I then looked up the attribute on [Mozilla](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-valuetext_attribute) and saw that it only applies to specific roles: progressbar, scrollbar, slider & spinbutton.

Just to try it out I went ahead and modified the role for the input element to slider, and sure enough VoiceOver then read out "Hello" instead of all of the underlines and whatnot.

So, I don't imagine you want to alter the role for these inputs, but it appears that VoiceOver at least will only look for the aria-valuetext property if the inputs role is one of the specified values in the spec.

**Comment by RobinHerbots** [comment]
@stuartclan , 

I was mistaken.  You should have tested with aria-label atttribute.

**Comment by RobinHerbots** [comment]
@stuartclan, 

I didn't realize I can enable the feature on my phone.  So now I can experiment with it.

**Comment by stuartclan** [comment]
@RobinHerbots,

Unfortunately "aria-label" simply helps define the field - I also tried aria-labelledby and aria-described-by. None of these alter VoiceOver's desire to read the contents of the input value...

I looked at the various aria options and I don't really see one that works for this purpose - the aria-valuetext seemed like the best shot at it.

I'm thinking using the input's value may simply not be possible while using showMaskOnFocus unless you implement some other element that is overlaying or behind and not being read-out by the focused input field.
